src/cursor/mod.rs

```diff
@@ -90,10 +91,7 @@ pub(crate) use common::{
 /// # }
 /// ```
 #[derive(Debug)]
-pub struct Cursor<T>
-where
-    T: DeserializeOwned + Unpin + Send + Sync,
```

Comment:
Since we no longer buffer `T`, we can drop the `Unpin + Send + Sync` parts of this, and because we now support borrowed deserialization, we can drop `DeserializeOwned` too.